pkg/istio/rbac/common/util.go

```
@@ -69,3 +272,152 @@ func NewConfig(configType string, namespace string, name string, spec proto.Mess
+	authzPolicyName := item.Resource.ConfigMeta.Name
+	namespace := item.Resource.ConfigMeta.Namespace
+	yamlFileName := authzPolicyName + "--" + namespace + ".yaml"
+	if _, err := os.Stat(localDirPath + yamlFileName); os.IsNotExist(err) {
+		log.Infof("file %s does not exist in local directory", localDirPath+yamlFileName)
+		return nil
+	} else if err != nil {
+		return fmt.Errorf("error stating file %s in local directory, error: %s", localDirPath+yamlFileName, err)
+	}
+	log.Infof("deleting file under path: %s\n", localDirPath+yamlFileName)
+	return os.Remove(localDirPath + yamlFileName)
+}
+
+// ReadConvertToModelConfig reads in the authorization policy yaml object and converts it into a model.Config struct
+func ReadConvertToModelConfig(serviceName, namespace, localDirPath string) (*model.Config, error) {
+	// define istio object interface to unmarshall yaml object into
+	item := &crd.IstioKind{Spec: map[string]interface{}{}}
+	yamlFileName := serviceName + "--" + namespace + ".yaml"
+	yamlFile, err := ioutil.ReadFile(localDirPath + yamlFileName)
+	if err != nil {
+		return nil, fmt.Errorf("unable to read yaml file to local directory, err: %s", err)
+	}
+	err = yaml.Unmarshal(yamlFile, item)
+	if err != nil {
+		return nil, fmt.Errorf("unable to unmarshal yaml file, err: %s", err)
+	}
+	config, err := crd.ConvertObject(collections.IstioSecurityV1Beta1Authorizationpolicies, item, "")
+	if err != nil {
+		return nil, fmt.Errorf("unable to convert yaml converted istio object to authorization policy model config, err: %s", err)
+	}
+	return config, nil
+}
+
+// FetchServicesFromDir walks through the files under a directory, based on the naming convention, it parses the service name
+// from the file name.
+func FetchServicesFromDir(namespace, localDirPath string) []string {
+	files, err := ioutil.ReadDir(localDirPath)
+	if err != nil {
+		log.Errorf("unable to read yaml files to local directory, err: %s", err)
```

Should we return an error here?